mirrord/cli/src/internal_proxy.rs

```diff
@@ -114,7 +145,19 @@ pub(crate) async fn proxy(args: InternalProxyArgs) -> Result<()> {
     // Create a main connection, that will be held until proxy is closed.
     // This will guarantee agent staying alive and will enable us to
     // make the agent close on last connection close immediately (will help in tests)
-    let (_main_connection, operator) = connect_and_ping(&config).await?;
+    let (main_connection, operator) = connect_and_ping(&config).await?;
```

Comment:
I fear after some time `main_connection` dies because there are no messages sent via it (maybe we should add a timer that sends pings while the proxy is still alive)